source/UF/SequentialColimits.lagda

```diff
@@ -0,0 +1,532 @@
+       V : (G' : ((n : ℕ) → b n ∼ b (succ n) ∘ a n))
+         → ((n : ℕ) → G n ∼ G' n) ≃ (G ＝ G')
+       V G' = ((n : ℕ) → G n ∼ G' n)
+                                      ≃⟨ Π-cong fe fe
+                                          (λ n → ≃-sym (≃-funext fe (G n) (G' n))) ⟩
+              ((n : ℕ) → G n ＝ G' n)
+                                      ≃⟨ ≃-sym (≃-funext fe G G') ⟩
+              (G ＝ G')               ■
+       VI = Σ-cong V
+       VII = singleton-≃-𝟙 (singleton-types-are-singletons G)
+
+ sequential-cocone-identity-characterization
+  : (𝓑 𝓑' : sequential-cocone 𝓐 B)
+  → (𝓑 ＝ 𝓑') ≃ (sequential-cocone-identity 𝓑 𝓑')
+ sequential-cocone-identity-characterization 𝓑 𝓑' =
+  (id-to-sequential-cocone-family 𝓑 𝓑' ,
+    Yoneda-Theorem-forth 𝓑 (id-to-sequential-cocone-family 𝓑)
+     (sequential-cocone-family-is-identity-system 𝓑) 𝓑')
+
+ sequential-cocone-family-to-id : (𝓑 𝓑' : sequential-cocone 𝓐 B)
+                                → (sequential-cocone-identity 𝓑 𝓑')
+                                → 𝓑 ＝ 𝓑'
+ sequential-cocone-family-to-id 𝓑 𝓑'
+  = ⌜ sequential-cocone-identity-characterization 𝓑 𝓑' ⌝⁻¹
+
+\end{code}
+
+Given a sequential cocone over X and a map X → Y there is a canonical assignment to
+a sequential cocone over Y.
+
+\begin{code}
+
+module _ (𝓐 : type-sequence 𝓤)
+         (X : 𝓥 ̇) (Y : 𝓣 ̇)
+       where
+
+ canonical-map-to-sequential-cocone : sequential-cocone 𝓐 X
+                                    → (X → Y)
+                                    → sequential-cocone 𝓐 Y
+ canonical-map-to-sequential-cocone (h , H) u =
+  ((λ n → u ∘ h n) , λ n → ∼-ap-∘ u (H n))
+
+\end{code}
+
+A sequential cocone over X is universal if the above map is an equivalence for any Y.
+Such a sequential cocone is said to be the sequential colimit of a type sequence.
+
+\begin{code}
+
+ Sequential-Colimit-Universal-Property : (𝓧 : sequential-cocone 𝓐 X)
+                                       → 𝓤 ⊔ 𝓥 ⊔ 𝓣 ̇
+ Sequential-Colimit-Universal-Property 𝓧 =
+  is-equiv (canonical-map-to-sequential-cocone 𝓧)
+
+\end{code}
+
+We now give a construction of the sequential colimit in terms of the pushout.
+This construction follows 26.2 in Introduction to Homotopy Type Theory (link above).
+
+The sequential colimit A∞ can be constructed as the pushout of the following diagram
+
+                     [id , σ]
+          Σ A + Σ A ------------> Σ A
+              |                    |
+   [id , id]  |                    | inrr
+              |                    |
+             Σ A ----------------> A∞
+                       inll
+
+where σ (n , x) = (n + 1 , a n x).
+
+\begin{code}
+
+module _ (𝓐@(A , a) : type-sequence 𝓤)
+         (X : 𝓣 ̇)
+       where
+
+ σ : Σ A → Σ A
+ σ (n , x) = (succ n , a n x)
+
+ f : Σ A + Σ A → Σ A
+ f = cases id id
+
+ g : Σ A + Σ A → Σ A
+ g = cases id σ
+
+ private
+  index : Σ A → ℕ
+  index = pr₁
+
+  element-at : ((n , x) : Σ A) → A n
+  element-at = pr₂
+
+ module _ (push-ex : pushouts-exist f g)
```

Comment:
The name `pushouts-exist` is not very good because it actually says that the pushout (singular) of the given span `(f,g)` exists.

Reply:
Should I reopen the pushout PR to change this?

Reply:
It can be changed as part of this PR.